uiowa/uiowa
# Facilities building content type, view and api

Resolves https://github.com/uiowa/uiowa/issues/5659

# To Test

- `ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local uli /node/add/building`
- Create a fake building (e.g. building number = asdf), then create a real building (e.g. 0002, Schaeffer Hall) with a custom name and a blank abbreviation.
- `ddev drush @facilities.local fm-buildings` (383 buildings created, 1 update, 1 deleted)
- https://facilities.uiowa.ddev.site/admin/content?title=&type=building&status=All&field_tags_target_id=
- Visit your real building (e.g. https://facilities.uiowa.ddev.site/building/0002), it should have correct information.
- Visit your fake building (e.g. https://facilities.uiowa.ddev.site/building/asdf), it should be a 404.
- Unlike other content types we create this one will be different in terms of our "content type checklist":
  - As a webmaster, you should be able to see the buildings in the content overview, but you should not have add/edit/delete permissions. Since this content type does not have editor interaction, moderation was not added.
  - On the building node itself you can see the revisions tab and for the one that was updated you should see more than one revision. There is a max of three revisions set.
  - There is basic meta tag/schema type Place information that is in the source, but that there isn't a body/teaser field to fill out meta descriptions.
- Go to a basic page and add the "Building list" views block under Site custom within the More... section. See that the filter works for building name, number and abbreviation. Table responsive is added.
- Test the `bypass node access` permission change by making sure a webmaster can assign a page to the menu system under an unpublished node. Also for the ambitious, you can check all of the custom content types listed below to make sure the webmasters can add/edit/delete them (exception being this building content type on facilities)

<details>
<summary>content types splits</summary>

  config/default/node.type.page.yml
  config/default/node.type.person.yml
  config/default/node.type.article.yml
  config/features/event/node.type.event.yml
  config/features/scholarship/node.type.scholarship.yml
  config/features/topic_page/node.type.topic_collection.yml
  config/features/areas_of_study/node.type.area_of_study.yml
  config/features/periodical/node.type.publication_issue.yml
  config/features/thesis_defense/node.type.thesis_defense.yml
  
  config/sites/hr.uiowa.edu/node.type.faq.yml
  config/sites/now.uiowa.edu/node.type.contact.yml
  config/sites/grad.uiowa.edu/node.type.mentor.yml
  config/sites/grad.uiowa.edu/node.type.scholar.yml
  config/sites/grad.uiowa.edu/node.type.student_profile.yml
  config/sites/uipress.uiowa.edu/node.type.book.yml
  config/sites/brand.uiowa.edu/node.type.lockup.yml
  config/sites/icsa.uiowa.edu/node.type.project.yml
  config/sites/iisc.uiowa.edu/node.type.partner.yml
  config/sites/iisc.uiowa.edu/node.type.project.yml
  config/sites/classrooms.uiowa.edu/node.type.room.yml
  config/sites/admissions.uiowa.edu/node.type.major.yml
  config/sites/admissions.uiowa.edu/node.type.event_series.yml
  config/sites/admissions.uiowa.edu/node.type.area_of_study.yml
  config/sites/admissions.uiowa.edu/node.type.transfer_tips.yml
  config/sites/admissions.uiowa.edu/node.type.student_profile.yml
  config/sites/policy.clas.uiowa.edu/node.type.book.yml
  config/sites/sppa.uiowa.edu/node.type.opportunity.yml
  config/sites/housing.uiowa.edu/node.type.residence_hall.yml
  config/sites/facilities.uiowa.edu/node.type.named_building.yml
  config/sites/grad.admissions.uiowa.edu/node.type.area_of_study.yml
  config/sites/grad.admissions.uiowa.edu/node.type.costs_estimate.yml
  config/sites/ighn.international.uiowa.edu/node.type.ighn_person.yml

</details>

There is a troublesome node: 321 Melrose Avenue (0307) where the abbreviation is different from the others. It is a blank string instead of NULL. Created a temporary check for it as well as a follow-up todo https://github.com/uiowa/uiowa/issues/6084
